File: app/src/main/java/com/vsa/filmoteca/model/sharedpreferences/SharedPreferencesManager.java
package com.vsa.filmoteca.model.sharedpreferences;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by seldon on 27/03/15.
 */
public class SharedPreferencesManager {

    //SharedPreferences
    private static final String SHAREDPREFERENCES_NAME = "MySharedPreferences";
    private static final String SHAREDPREFERENCES_CURRENT = "Current";
    private static final String SHAREDPREFERENCES_SIZE = "Size";

    private static SharedPreferences mSharedPreferences;

    private static void initSharedPreferences(Context context) {
        if(mSharedPreferences == null)
            mSharedPreferences = context.getSharedPreferences(SHAREDPREFERENCES_NAME, Context.MODE_PRIVATE);
    }

    public static int getCurrentMovieIndex(Context context) {
        initSharedPreferences(context);
        return mSharedPreferences.getInt(SHAREDPREFERENCES_CURRENT, 0);
    }

    public static void setCurrentMovieIndex(Context context, int index) {
        initSharedPreferences(context);
        mSharedPreferences
                .edit()
                .putInt(SHAREDPREFERENCES_CURRENT, index)
                .commit();
    }

    public static int getMoviesCount(Context context) {
        initSharedPreferences(context);
        return mSharedPreferences.getInt(SHAREDPREFERENCES_SIZE, 0);
    }

    public static void setMoviesCount(Context context, int index) {
        initSharedPreferences(context);
        mSharedPreferences
                .edit()
                .putInt(SHAREDPREFERENCES_SIZE, index)
                .commit();
    }

}

File: app/src/main/java/com/vsa/filmoteca/data/source/sharedpreferences/SharedPreferencesManager.kt
package com.vsa.filmoteca.data.source.sharedpreferences

import android.content.Context
import android.content.SharedPreferences
import javax.inject.Inject

/**
 * Created by seldon on 27/03/15.
 */
class SharedPreferencesManager @Inject constructor(context: Context) {

    companion object {
        private const val KEY_NAME = "MySharedPreferences"
        private const val KEY_CURRENT = "Current"
        private const val KEY_SIZE = "Size"
        private const val KEY_REVIEW_ALREADY_SUGGESTED = "review_already_suggested"
        private const val KEY_FIRST_EXECUTION_TIME_MILLIS = "first_execution_time_millis"
        private const val KEY_APP_VISITS = "app_visits"
    }

    private val sharedPreferences: SharedPreferences = context.getSharedPreferences(KEY_NAME, Context.MODE_PRIVATE)

    var currentMovieIndex: Int
        get() = sharedPreferences.getInt(KEY_CURRENT, 0)
        set(index) {
            sharedPreferences
                    .edit()
                    .putInt(KEY_CURRENT, index)
                    .apply()
        }

    var moviesCount: Int
        get() = sharedPreferences.getInt(KEY_SIZE, 0)
        set(index) {
            sharedPreferences
                    .edit()
                    .putInt(KEY_SIZE, index)
                    .apply()
        }

    var isReviewAlreadySuggested: Boolean
        get() = sharedPreferences.getBoolean(KEY_REVIEW_ALREADY_SUGGESTED, false)
        set(value) {
            sharedPreferences.edit()
                    .putBoolean(KEY_REVIEW_ALREADY_SUGGESTED, value)
                    .apply()
        }

    var fistExecutionTimeMillis: Long
        get() = sharedPreferences.getLong(KEY_FIRST_EXECUTION_TIME_MILLIS, -1)
        set(value) {
            sharedPreferences.edit()
                    .putLong(KEY_FIRST_EXECUTION_TIME_MILLIS, value)
                    .apply()
        }

    var appVisitsCounter: Int
        get() = sharedPreferences.getInt(KEY_APP_VISITS, 0)
        set(value) {
            sharedPreferences.edit()
                    .putInt(KEY_APP_VISITS, value)
                    .apply()
        }

}
